src/lib/Libcmds/ck_job_name.c

```diff
@@ -103,7 +103,7 @@ int   chk_alpha;
 	}
 
 	while (*name) {
-		if (isgraph((int)*name++) == 0)	/* disallow any non-printing */
+		if (isalnum((int)*name++) == 0)	/* disallow any non-printing */
```

Comment:
This could potentially cause regressions for customers that currently use certain special characters in their job names. My interpretation of the passage in the reference guide is that the following characters are legal in a job name (because they don't conflict with filesystem operations):
- Alphabetic (both upper and lower case)
- Numeric
- Plus sign ( + )
- Minus sign ( - )
- Underscore ( _ )

All other "special" characters (which had previously been legal) will become illegal, including but not limited to:
- ~
- !
- \#
- $
- %
- ^
- &
- \*
- (
- )
- {
- }
- =
- [
- ]
- colon
- "
- =
- /
- \
- |
- ?

So if anyone currently names their job with any of these characters, it will soon be rejected by PBS Pro. Are there other characters like equal and period that we should also allow? Seems like we're going from a very permissive approach to a very restrictive one, and I don't want it to bite us.

Also, if we're upgrading to a newer version of PBS Pro that is more restrictive than the old version in terms of job names, do we need to check the names of queued and running jobs or are we safe to ignore them?

Reply:
Thanks @mike0042 for review.
I changed the the function in my next commit from isalnum() to isalnumspch(), which will allow alphanumeric characters, Plus sign ( + ), Minus sign ( - ), Underscore ( _ ) .
I have tested special characters in job name, qsub feature in "pbs_version = 19.0.0". Where its allowing special characters (like &, / etc ) in job name both via command line and job script.
Do you prefer any other version where I should check this?